internal/gw/client_publish_errors_test.go

```diff
@@ -123,17 +123,19 @@ func TestClientPublishErrors(t *testing.T) {
 		publish.raw.Links["self"] = "/publish/1234"
 
 		gw.nextHTTPResponse = &http.Response{
-			Status:     "418 I'm a teapot",
-			StatusCode: 418,
-			Body:       io.NopCloser(strings.NewReader("")),
+			Status:     "409 Conflict",
+			StatusCode: 409,
+			Body: io.NopCloser(strings.NewReader(
+				"{\"detail\": \"Publish in unexpected state\"}",
+			)),
 		}
 
 		err := publish.AddItems(ctx, []ItemInput{{"/some/uri", "abc123"}})
 
 		if err == nil {
 			t.Error("Unexpectedly failed to return an error")
 		}
-		if !strings.Contains(err.Error(), "I'm a teapot") {
+		if !strings.Contains(err.Error(), "Conflict") {
```

Comment:
As the main point of the change is to start parsing the error body (if there was one), should this be tweaked or extended so it's looking for "Publish in unexpected state" instead?

Right now it seems as though this updated test would pass even without the change to client.go.